### ISSUE TITLE
Find and display the longest path

### ISSUE THREAD

**Issue by nivdror** [apache/airflow#39938]
### Description

Create a view in the graph UI by pressing two tasks that are dependent on one another and show the longest path and how much time it will take(by using task duration over the recent history).

We created this feature in our own old airflow but I don't see why this cannot be in apache airflow.
Here is an example:

![Screenshot 2024-05-30 at 11 47 45](https://github.com/apache/airflow/assets/131352704/62a1a2e5-33ef-439f-99af-efc36f44d1da)

### Use case/motivation

I'm trying to achieve a better visibility over the graph UI which I believe is the important UI in airflow.
This feature will help the oncalls figure out what the ETA of a certain flow

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
@nivdror Are you intend to make a Pull Request for add this feature into the Grid View?

**Comment by nivdror** [comment]
No, the grid view is of lesser importance to me and fellow to my data engineers (at least in my company)

**Comment by Taragolis** [comment]
Old Graph UI was removed in [Airflow 2.7](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#old-graph-view-is-removed-32958), and moved into the Grid View.

**Comment by nivdror** [comment]
while I think removing the old graph UI was a mistake. what done is done.
please forgive my choice of words. what I meant for that I like it to be added into the Graph view in the Grid UI.
